Posh-ACME/Plugins/EfficientIP.ps1

```diff
@@ -0,0 +1,237 @@
+
+        $DNSRRRecord = Get-EIPDNSRecordID -Name $RecordName -TxtValue $TxtValue -DNSName $EIPDNSName -View $EIPView -EIPUsername $EIPUsername -EIPPassword $EIPPasswordInsecure -EIPHostname $EIPHostname
+        If($DNSRRRecord.count -eq 0){
+           Return
+        }
+    
+        $Response = $DNSRRRecord | % { 
+            $ID = $_
+            $Endpoint = "dns_rr_delete"
+            $Parameters = "rr_id=$($ID)"
+            $Response = Send-EfficientIPRequest -Parameters $Parameters -Endpoint $Endpoint -Method DELETE -EIPUsername $EIPUsername -EIPPassword $EIPPasswordInsecure -EIPHostname $EIPHostname
+        }
+
+    }Catch{ }
+
+    <#
+    .SYNOPSIS
+        Remove a DNS TXT record from EfficientIP SOLIDServer.
+
+    .DESCRIPTION
+        Remove a DNS TXT record from EfficientIP SOLIDServer.
+
+    .PARAMETER RecordName
+        The fully qualified name of the TXT record.
+
+    .PARAMETER EIPUsername
+        The EfficientIP SOLIDServer Username.
+
+    .PARAMETER EIPPassword
+        The EfficientIP SOLIDServer Password.
+
+    .PARAMETER EIPHostname
+        The EfficientIP SOLIDServer Hostname.
+
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+
+    .EXAMPLE
+        $secret = Read-Host "EIP Password" -AsSecureString
+        PS C:\>Remove-DnsTxt '_acme-challenge.example.com' 'user' $secret 'eip.local'
+
+        Removes a TXT record for the specified site with the specified value.
+    #>
+}
+
+function Save-DnsTxt {
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromRemainingArguments)]
+        $ExtraParams
+    )
+    <#
+    .SYNOPSIS
+        Not required.
+
+    .DESCRIPTION
+        This provider does not require calling this function to commit changes to DNS records.
+
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+    #>
+}
+
+############################
+# Helper Functions
+############################
+
+Function Send-EfficientIPRequest {
+   Param(
+      [Parameter(Mandatory,Position=0)][String]$EIPHostname,
+      [Parameter(Mandatory,Position=1)][String]$EIPUsername,
+      [Parameter(Mandatory,Position=2)][String]$EIPPassword,
+      [Parameter(Mandatory,Position=3)][String]$Endpoint,
+      [Parameter(Mandatory,Position=4)][String]$Method = "Get",
+      [Parameter(Mandatory,Position=5)][String]$Parameters
+   )
+
+   begin {}
+
+   process {
+      $basicAuthValue = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(("{0}:{1}" -f $EIPUsername, $EIPPassword)))
+      $Headers = @{
+         "Authorization"="Basic $basicAuthValue"
+         "Content-Type"="application/json"
+         "Accept"="application/json"
+         "charset"="utf-8"
+      }
+      $URI = "/rest/$($Endpoint)?$($Parameters)"
+      $URL = "https://$($EIPHostname)$($URI)"
```

Comment:
Notice that unfortunately many still rely on self signed certificate. I hope it's possible to disable SSL certificat and hostname checks.